File: interviewbit/math/excel-column-number.java
/*
    interviewbit
    track: "programming",
    section: "math",
    problem title: "Excel Column Number"
*/
public class Solution {
    public int titleToNumber(String A) {
        int baseCharValue = 64;
        int colNo = 0;
        for(int i=A.length()-1,j=1;i>=0;i--,j++){
            // colNo+=(((int)A.charAt(i))-64) + (i*26);
            if(j-1==0){
                colNo+= (((int)A.charAt(i))-64);
            } else{
                colNo+= (((int)A.charAt(i))-64)*Math.pow(26,(j-1));
            }
            // colNo+=(((int)A.charAt(i))-64)+((j-1)*26);
        }
        return colNo;
    }
}

File: interviewbit/math/excel-column-number.js
/*
	interviewbit
	track: "programming"
	section: "math"
	problem title: "Excel Column Number"
*/

module.exports = { 
 //param A : string
 //return an integer
	titleToNumber : function(A){
	    var num = 0;
	    for(var i=A.length-1;i>=0;i--){
	        num+=(A.charCodeAt((A.length-1)-i)-64)*(Math.pow(26,i));
	    }
	    return num;
	}
};
